homeassistant/components/wemo/wemo_device.py

```diff
@@ -0,0 +1,89 @@
+"""Home Assistant wrapper for a pyWeMo device."""
+import logging
+
+from pywemo import PyWeMoException, WeMoDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_DEVICE_ID,
+    CONF_NAME,
+    CONF_PARAMS,
+    CONF_TYPE,
+    CONF_UNIQUE_ID,
+)
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN, WEMO_SUBSCRIPTION_EVENT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DeviceWrapper:
+    """Home Assistant wrapper for a pyWeMo device."""
+
+    def __init__(self, hass: HomeAssistant, wemo: WeMoDevice, device_id: str) -> None:
+        """Initialize DeviceWrapper."""
+        self.hass = hass
+        self.wemo = wemo
+        self.device_id = device_id
+        self.device_info = _device_info(wemo)
+        self.supports_long_press = wemo.supports_long_press()
+
+    def _subscription_callback(
+        self, _device: WeMoDevice, event_type: str, params: str
+    ) -> None:
+        self.hass.bus.fire(
+            WEMO_SUBSCRIPTION_EVENT,
+            {
+                CONF_DEVICE_ID: self.device_id,
+                CONF_NAME: self.wemo.name,
+                CONF_TYPE: event_type,
+                CONF_PARAMS: params,
+                CONF_UNIQUE_ID: self.wemo.serialnumber,
+            },
+        )
+
+
+def _device_info(wemo: WeMoDevice):
+    return {
+        "name": wemo.name,
+        "identifiers": {(DOMAIN, wemo.serialnumber)},
+        "model": wemo.model_name,
+        "manufacturer": "Belkin",
+    }
+
+
+async def async_register_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, wemo: WeMoDevice
+) -> DeviceWrapper:
+    """Register a device with home assistant and enable pywemo event callbacks."""
+    device_registry = await hass.helpers.device_registry.async_get_registry()
```

Comment:
Please import and use `async_get` from the device registry helper instead.

Reply:
Done